src/html-media-source.js

```diff
@@ -231,10 +231,22 @@ export default class HtmlMediaSource extends videojs.EventTarget {
     let buffer;
     let parsedType = parseContentType(type);
 
-    // create a virtual source buffer to transmux MPEG-2 transport
+    // Create a VirtualSourceBuffer to transmux MPEG-2 transport
     // stream segments into fragmented MP4s
     if (parsedType.type === 'video/mp2t') {
-      // default codecs
+      if (this.sourceBuffers.length !== 0) {
+        // If another VirtualSourceBuffer already exists, then we are creating a
+        // SourceBuffer for an alternate audio track and therefore we know that
+        // the source has both an audio and video track.
+        // That means we should trigger the manual creation of the real
+        // SourceBuffers instead of waiting for the transmuxer to return data
```

Comment:
Is this comment inaccurate or are we forbidding creating an audio VSB *first* and then creating the video VSB?